antlr-kotlin-runtime/src/commonMain/kotlin/org/antlr/v4/kotlinruntime/tree/Trees.kt

```diff
@@ -76,24 +76,19 @@ public object Trees {
 
   public fun getNodeText(t: Tree, ruleNames: List<String>?): String {
     if (ruleNames != null) {
-      if (t is RuleContext) {
-        val ruleIndex = t.ruleContext.ruleIndex
-        val ruleName = ruleNames[ruleIndex]
-        val altNumber = t.altNumber
-
-        return if (altNumber != ATN.INVALID_ALT_NUMBER) {
-          "$ruleName:$altNumber"
-        } else {
-          ruleName
-        }
-      } else if (t is ErrorNode) {
-        return t.toString()
-      } else if (t is TerminalNode) {
-        val symbol = t.symbol
-
-        if (symbol != null) {
-          return symbol.text!!
+      when (t) {
```

Comment:
Should we factor out the return?

Reply:
We cannot as that's a non-local return (`when` inside of an `if`). Will be supported starting from Kotlin 2.0.

Reply:
Sorry, wrong answer. We cannot, as it requires an `else` branch to return something. We don't want that, we want it to continue over the rest of the method, as it was doing with if-else.